llvm/include/llvm/IR/Cheri.h

```diff
@@ -55,6 +56,68 @@ inline bool isCheriPointer(Type *Ty, const DataLayout *DL) {
 }
 
 namespace cheri {
+
+/// Represents the LLVM-level attribute that is used to signal
+/// cross-compartment imports.
+class CapabilityImportAttr {
+public:
+  /// Parses the string value previously constructed into an instance of the
+  /// attribute object.
+  CapabilityImportAttr(StringRef ValueRef) {
+    size_t NextSubstrStartsAt = 0;
+
+    ExportPrefix = ValueRef.substr(
+        NextSubstrStartsAt, ValueRef.find(SEPARATOR) - NextSubstrStartsAt);
+
+    NextSubstrStartsAt += ExportPrefix.size() + 1;
+    if (NextSubstrStartsAt < ValueRef.size()) {
+      Domain = ValueRef.substr(NextSubstrStartsAt,
+                               ValueRef.find(SEPARATOR, NextSubstrStartsAt) -
+                                   NextSubstrStartsAt);
+    } else {
+      Domain = StringRef();
+    }
+
+    NextSubstrStartsAt += Domain.size() + 1;
+    if (NextSubstrStartsAt < ValueRef.size()) {
+      ObjectName = ValueRef.substr(
+          NextSubstrStartsAt,
+          ValueRef.find(SEPARATOR, NextSubstrStartsAt) - NextSubstrStartsAt);
+    } else {
+      ObjectName = StringRef();
+    }
+
+    NextSubstrStartsAt += ObjectName.size() + 1;
+    if (NextSubstrStartsAt < ValueRef.size()) {
+      EncodedPermissions = ValueRef.substr(NextSubstrStartsAt);
+    } else {
+      EncodedPermissions = StringRef();
+    }
+  }
+
+  CapabilityImportAttr(StringRef Domain, StringRef ObjectName,
+                       StringRef EncodedPermissions,
+                       StringRef ExportPrefix = StringRef(""))
+      : Domain(Domain), ObjectName(ObjectName),
+        EncodedPermissions(EncodedPermissions), ExportPrefix(ExportPrefix) {};
+
+  /// Returns the name of attribute.
+  static const StringRef getAttrName();
+
+  /// Returns the string value for this CapabilityImportAttr.
+  std::string str();
+
+  StringRef Domain;
+  StringRef ObjectName;
+  StringRef EncodedPermissions;
+  StringRef ExportPrefix;
+
+private:
+  /// The separator used when creating the final string value for the
+  /// attribute.
+  static constexpr char SEPARATOR = ',';
```

Comment:
I would have this as a `char*` so that you don't need `&SEPARATOR` at each use.

Reply:
https://github.com/CHERIoT-Platform/llvm-project/pull/147/commits/4ffcc2cb376e395259065c8636b29833bfce7139#diff-9bdc401a1ee7793838a466dbec73b4cd8ce3a57e222b19e06fc9d429d2c561c3R197 makes it a `char*`.